source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -604,6 +611,37 @@ def _getAvailableAddonsInBG(self):
 		core.callLater(delay=0, callable=self.detailsVM.updated.notify, addonDetailsVM=self.detailsVM)
 		log.debug("completed refresh")
 
+	def cancelInstallAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		log.debug(f"Cancelling install of {listItemVM.Id}")
+
+		futuresCopy = self._downloader._pending.copy()
+		try:
+			addonDataManager._downloadsPendingCompletion.remove(listItemVM)
+			for future, addon in futuresCopy.items():
+				if listItemVM == addon[0] and not future.cancel():
+					self._downloader.progress.pop(addon[0], None)
+			listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
+			return
+		except KeyError:

```

Comment:
is this KeyError just for the line `addonDataManager._downloadsPendingCompletion`? if so, the rest of the code in the `try` statement should be moved to `else` after `except`